pkgs/test/by-name/src/eval.rs

```diff
@@ -0,0 +1,114 @@
+use crate::structure;
+use crate::utils::ErrorWriter;
+
+use anyhow::Context;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::ffi::OsString;
+use std::io;
+use std::path::PathBuf;
+use std::process;
+use tempfile::NamedTempFile;
+
+/// Attribute set of this structure is returned by eval.nix
+#[derive(Deserialize)]
+struct AttributeInfo {
+    call_package_file: Option<PathBuf>,
+    is_derivation: bool,
+}
+
+const EXPR: &str = include_str!("eval.nix");
+
+pub fn check_values<W: io::Write>(
+    writer: &mut ErrorWriter<W>,
+    nixpkgs: &structure::Nixpkgs,
+    extra_nix_path: &OsStr,
+) -> anyhow::Result<()> {
+    let attrs_file = NamedTempFile::new().context("Failed to create a temporary file")?;
+    serde_json::to_writer(&attrs_file, &nixpkgs.package_names).context(format!(
+        "Failed to serialise the package names to the temporary path {}",
+        attrs_file.path().display()
+    ))?;
+
+    let mut nix_path = OsString::new();
+    nix_path.push(attrs_file.path());
+    nix_path.push(":");
+    nix_path.push(&nixpkgs.path);
+    if !extra_nix_path.is_empty() {
+        nix_path.push(":");
+        nix_path.push(extra_nix_path);
+    }
+
+    let mut command = process::Command::new("nix-instantiate");
+    command
+        .stderr(process::Stdio::inherit())
+        .env("NIX_PATH", nix_path)
+        .args([
+            "--eval",
+            "--json",
+            "--strict",
+            "--readonly-mode",
+            "--restrict-eval",
+            "--show-trace",
+            "--expr",
+            EXPR,
+        ])
+        .arg("-I")
+        .arg(attrs_file.path())
+        .args(["--arg", "attrsPath"])
+        .arg(attrs_file.path())
+        .arg("-I")
+        .arg(&nixpkgs.path)
+        .args(["--arg", "nixpkgsPath"])
+        .arg(&nixpkgs.path);
+
+    let result = command
+        .output()
+        .context(format!("Failed to run command {command:?}"))?;
+
+    if !result.status.success() {
+        anyhow::bail!("Failed to run command {command:?}");
+    }
+    let actual_files: HashMap<String, AttributeInfo> = serde_json::from_slice(&result.stdout)
+        .context(format!(
+            "Failed to deserialise {}",
+            String::from_utf8_lossy(&result.stdout)
+        ))?;
+
+    for package_name in &nixpkgs.package_names {
+        let relative_package_file = structure::Nixpkgs::relative_file_for_package(package_name);
+        let absolute_package_file = nixpkgs.path.join(&relative_package_file);
+
+        if let Some(attribute_info) = actual_files.get(package_name) {
+            let is_expected_file =
+                if let Some(call_package_file) = &attribute_info.call_package_file {
+                    &absolute_package_file == call_package_file
```

Comment:
Wouldn't:
```suggestion
                    absolute_package_file == *call_package_file
```

work?

Reply:
tests pass with this, and to the best of my knowledge are (in this specific context) equivalent.

Reply:
Nice, I'm not entirely sure about the difference, but it works!

Reply:
comparing `&X` vs comparing `X`

Reply:
if you have autoderef, it's comparing `X` in the end, but better safe than sorry in expressing what you really intend for

Reply:
(also it's useless to have `*&X == *Y` rather than `X == *Y` :P)